contracts/test/umbra-gsn-test.js

```diff
@@ -39,24 +38,29 @@ describe('Umbra GSN', () => {
     ctx.token = await TestToken.new('TestToken', 'TT');
     await ctx.token.mint(payer, tokenAmount);
 
-    // Start the GSN Test environment— this includes deployment of a relay hub, a forwarder, and
-    // a stake manager, as well as starting a relay server. It also deploys a naive Paymaster, but we
-    // will use our own
-    const gsnInstance = await GsnTestEnvironment.startGsn(
-      'localhost'
-    );
+    // Start the GSN Test environment— this includes deployment of a relay hub and
+    // a stake manager, as well as starting a relay server. It also deploys a naive Paymaster and
+    // a Forwarder implementation, but in both cases, we will use our own
+    const gsnInstance = await GsnTestEnvironment.startGsn('localhost');
 
     // Deploy our custom forwarder, which ignore signature validation, and save its address
     // We'll need it when sending contract calls via our RelayProvider
     const deployedForwarder = await UmbraForwarder.new({ from: owner });
     ctx.forwarder = deployedForwarder.address;
 
-    // Deploy the Umbra GSN wrapper & paymaster
+    // Deploy the Umbra GSN wrapper
     ctx.relayRecipient = await UmbraRelayRecipient.new(ctx.umbra.address, ctx.forwarder, {
       from: owner,
     });
+
+    // Deploy the Umbra Paymaster
     ctx.paymaster = await UmbraPaymaster.new(ctx.relayRecipient.address, { from: owner });
```

Comment:
Didn't we migrate to hardhat? It looks like we still use a lot of web3 in the tests. Maybe we didn't migrate to ethers-style and just changed the VM?

Reply:
Yes, we are using hardhat, and the hardhat-web3 plugin. Tests could be updated to use ethers, but it wasn't strictly necessary!